### DISCUSSION TITLE
Movement selection inconsistency?

### DISCUSSION THREAD

**Discussion by yusuphgammut** [helix-editor/helix#6033] [General] [answered]
Hi Helix team. I'm getting used to Helix motions and selections. I was wondering if a particular behavior of a motion + selection is expected or should be fixed. I'll use some screenshots to explain it:

**Step 1:** Initial position
<img width="333" alt="Screen Shot 2023-02-17 at 1 32 58 PM" src="https://user-images.githubusercontent.com/12637787/219767732-bfd193c2-61a8-425d-9164-02993134bb65.png">
**Step 2:** Press `b` to go backwards. Note that the word "second" gets selected.
<img width="333" alt="Screen Shot 2023-02-17 at 1 33 19 PM" src="https://user-images.githubusercontent.com/12637787/219767734-f9a4ae40-f139-4d5c-aa41-f7dc6a5fee10.png">
**Step 3:** Press `v` to start selection. I want to select from the start of the word "second" to the end of the word "fourth".
<img width="334" alt="Screen Shot 2023-02-17 at 1 33 45 PM" src="https://user-images.githubusercontent.com/12637787/219767735-8463dc05-d0ca-4058-b3de-136c40046da2.png">
**Step 4:** Press `w` to go forward. Note that the selection on the word "second" disappears.
<img width="333" alt="Screen Shot 2023-02-17 at 1 34 07 PM" src="https://user-images.githubusercontent.com/12637787/219767736-51e49a14-5183-4157-81a7-17ce3973ce0b.png">
**Step 5:** Press `w` two times. Note that the resulting selection starts from the end of the word "second".
<img width="334" alt="Screen Shot 2023-02-17 at 1 34 32 PM" src="https://user-images.githubusercontent.com/12637787/219767740-05a42a39-ed21-47d8-a2e4-72281e39cdfc.png">

My question is if in the example above, the resulting selection should start from the beginning of the word "second", or if this is the expected behavior. I was reading the philosophy around [Kakoune's editing model](https://kakoune.org/why-kakoune/why-kakoune.html#_improving_on_the_editing_model) and they say that *moving **is** selecting*. Considering that, the backwards motion in step 2 of this example should be considered selection, and the forward motions after pressing `v` should extend that selection. Thus the resulting selection after pressing `v` should be `second third fourth `.

Let me know your thought on this.

**Comment by the-mikedavis** [comment, marked as answer]
This behavior is intended. Ranges in selections have a head and an achor and in select mode the anchor stays put while the head is moved. (The head is where the cursor is.)

To select "second third fourth" in this case you would want to use `flip_selections` (`A-;`) after step 2 to swap the head and anchor.

Moving around doesn't necessarily always give you a larger selection. Near the end of that section in the Kakoune docs there is a note: "capital commands tend to expand the selection." (Where capital commands in Kakoune are generally the same as select-mode in Helix.) Motions might also be used to tune existing selections in select mode: you might use `h` to shrink a forwards-facing selection or `k` to shrink a block selection by one line for example.

**Reply by yusuphgammut** [reply]
Ok, I understand the logic now. It makes sense. Maybe it'll be worth mentioning this head/anchor thing somewhere in the docs. I can try to make a PR but I'll wait to have a bit more experience with the editor.
